From d0551277af81f6275a003efe4a9ff6f6d0201881 Mon Sep 17 00:00:00 2001
From: Wenyou Yang <wenyou.yang@atmel.com>
Date: Fri, 30 Jan 2015 11:27:34 +0800
Subject: [PATCH] Config: add Audio Codec menuconfig's condition

To decrease the menuconfig complexity, selecting the Audio Codec to
the power down mode is determined by the board, by default.

Signed-off-by: Wenyou Yang <wenyou.yang@atmel.com>
---
 Config.in                       | 5 +++--
 board/Config.in                 | 4 ++++
 board/sama5d4ek/Config.in.board | 1 +
 3 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/Config.in b/Config.in
index 1317479..6da69b0 100644
--- a/Config.in
+++ b/Config.in
@@ -435,9 +435,10 @@ config	CONFIG_HDMI_ON_TWI3
 endchoice
 
 config CONFIG_WM8904
-	bool "CODEC (wm8904)"
+	bool "Audio CODEC (wm8904)"
+	depends on CONFIG_HAS_AUDIO_CODEC
 	select CONFIG_TWI
-	default n
+	default y
 	help
 	  This interface allow you to select the CODEC chip on the board
 	  which will be put into the Power-Down mode.
diff --git a/board/Config.in b/board/Config.in
index 421d3bc..fce3a52 100644
--- a/board/Config.in
+++ b/board/Config.in
@@ -388,3 +388,7 @@ config CONFIG_HAS_EHT0_PHY
 config CONFIG_HAS_EHT1_PHY
 	bool
 	default n
+
+config CONFIG_HAS_AUDIO_CODEC
+	bool
+	default n
diff --git a/board/sama5d4ek/Config.in.board b/board/sama5d4ek/Config.in.board
index c5b6398..b9397ea 100644
--- a/board/sama5d4ek/Config.in.board
+++ b/board/sama5d4ek/Config.in.board
@@ -22,6 +22,7 @@ config CONFIG_SAMA5D4EK
 	select CONFIG_PM
 	select CONFIG_HAS_EHT0_PHY
 	select CONFIG_HAS_EHT1_PHY
+	select CONFIG_HAS_AUDIO_CODEC
 	select SUPPORT_BUS_SPEED_176MHZ
 	select SUPPORT_BUS_SPEED_170MHZ
 	select SUPPORT_BUS_SPEED_148MHZ
-- 
1.9.1

